### DISCUSSION TITLE
How to convert from m3u8 to project json format?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [iptv-org/community#255] [Q&A] [answered]
Do you have a script to convert a `m3u8` link which lists channels with metadata info (tvg-logo, group-title, etc) to the json format used in your project?

**Comment by BellezaEmporium** [comment, marked as answer]
Our "generate.js" file contain a function that generates a JSON : 

https://github.com/iptv-org/iptv/blob/c24ea7dfdaabcf5f31ff00cf01b314149f25d245/scripts/generate.js#L170

**Reply by pvonmoradi** [reply]
@LaneSh4d0w 
Thanks. Specifically, I think in your pipeline, there is a method to convert "amalgamated links" like https://iptv-org.github.io/iptv/index.country.m3u to a `channels.json`.  Do you know exactly what steps should I take to input the mentioned link and get a `channels.json`?

**Reply by BellezaEmporium** [reply]
I'll call @freearhey to better inform you about this subject. He's the one who created and in charge of all the scripts of this repository. He'll be able to explain it better than I do.

**Reply by freearhey** [reply]
Playlists are parsed using the [iptv-playlist-parser](https://github.com/freearhey/iptv-playlist-parser) package.

**Reply by pvonmoradi** [reply]
@freearhey Thanks. Still, there should be another script to convert `iptv-playlist-parser`'s output to project json...

**Reply by freearhey** [reply]
It takes place in [scripts/parser.js](https://github.com/iptv-org/iptv/blob/c24ea7dfdaabcf5f31ff00cf01b314149f25d245/scripts/parser.js)
